src/main/com/yetanalytics/datasim/input.clj

```diff
@@ -66,6 +67,35 @@
   (when-some [ed (s/explain-data ::parameters parameters)]
     (errs/explain-to-map-coll ::parameters ed)))
 
+(defn validate-pattern-filters
+  [{{:keys [gen-profiles
+            gen-patterns]} :parameters
+    :keys                  [profiles]}]
+  (let [profile-idset (into #{}
+                            (map :id profiles))
+        pattern-idset (into #{}
+                            (keep (fn [{:keys [id primary]}]
+                                    (when (and primary)
+                                      id))
+                                  (mapcat :patterns profiles)))]
+    (concat
+     (for [[idx profile-id] (map-indexed vector gen-profiles)
+           :when            (not (contains? profile-idset profile-id))]
+       {:id   (str "parameters-gen-profiles-" idx)
+        :path [:parameters :gen-profiles idx]
+        :text (format "Profile ID %s is not one of provided profiles: %s"
+                      profile-id
+                      (cs/join \, profile-idset))})
+     (for [[idx pattern-id] (map-indexed vector gen-patterns)
+           :when            (not (contains? pattern-idset pattern-id))]
+       {:id   (str "parameters-gen-patterns-" idx)
+        :path [:parameters :gen-patterns idx]
+        :text
+        (format
+         "Pattern ID %s is not among primary patterns in provided profiles: %s"
+         pattern-id
+         (cs/join \, pattern-idset))}))))
```

Comment:
I did indeed write this in line with what the `validate-profiles` function above does. When I look at the errors ns it is functions for transforming errors for format (like from spec, etc), not specific domain things like the content of validation for a given parameter. I don't think it is appropriate to move those there.

Reply:
Are you asking for like a _general_ function to make error maps that takes id, path and text? I guess I could see that for enforcing shape (as there is currently no spec for these errors), but just forming a map with literal syntax seems like a lot less lift than a function wrapping the same.

Reply:
I see what you mean. For now I feel like a _general_ function will be a bit overkill for just two error types. I guess the amount of clutter the `format` calls introduced caught my eye, so I immediately started thinking about ways to refactor this section.

In the future it would perhaps be nice to revisit this so that the error functions are a bit more consistent though. Adding specs would be nice as well.

Reply:
> In the future it would perhaps be nice to revisit this so that the error functions are a bit more consistent though. Adding specs would be nice as well.

Agreed, errors need some love and specs should very much exist, I'll put in a ticket for future work on that.